sccm-technicalpreview/azuredeploy.json

```diff
@@ -71,91 +78,68 @@
       "sourcePortRange": "*",
       "destinationAddressPrefix": "*"
     },
-    "vmrole": [
-      "DC",
-      "DPMP",
+    "vmrole": "[if(parameters('createclient'),createarray('DC','DPMP','Client'),createarray('DC','DPMP'))]",
+    "vmInfo": "[if(parameters('createclient'),json('{\"DC\":{\"name\":\"DC01\",\"DiskType\":\"Premium_LRS\",\"size\":\"Standard_B2s\",\"imageReference\":{\"publisher\": \"MicrosoftWindowsServer\",\"offer\": \"WindowsServer\",\"sku\": \"2019-Datacenter\",\"version\": \"latest\"}},\"DPMP\":{\"name\":\"DPMP01\",\"DiskType\":\"Premium_LRS\",\"size\":\"Standard_B2s\",\"imageReference\":{\"publisher\": \"MicrosoftWindowsServer\",\"offer\": \"WindowsServer\",\"sku\": \"2019-Datacenter\",\"version\": \"latest\"}},\"Client\":{\"name\":\"Cl01\",\"DiskType\":\"Premium_LRS\",\"size\":\"Standard_B2s\",\"imageReference\":{\"publisher\": \"MicrosoftWindowsDesktop\",\"offer\": \"Windows-10\",\"sku\": \"20h1-pro\",\"version\": \"latest\"}}}'),json('{\"DC\":{\"name\":\"DC01\",\"DiskType\":\"Premium_LRS\",\"size\":\"Standard_B2s\",\"imageReference\":{\"publisher\": \"MicrosoftWindowsServer\",\"offer\": \"WindowsServer\",\"sku\": \"2019-Datacenter\",\"version\": \"latest\"}},\"DPMP\":{\"name\":\"DPMP01\",\"DiskType\":\"Premium_LRS\",\"size\":\"Standard_B2s\",\"imageReference\":{\"publisher\": \"MicrosoftWindowsServer\",\"offer\": \"WindowsServer\",\"sku\": \"2019-Datacenter\",\"version\": \"latest\"}}}'))]",
```

Comment:
remove the inline json here - break it out into vars and then just use the vars in the if() statement.